### PR TITLE
Disallow additionalProperties on objects

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/roadmap/issues/2132

This PR sets `"additionalProperties": false` on the objects that don't have specific configuration for additional properties. This is to catch bad user input, e.g. the use of misspelled or outdated properties.

The `.cluster-shared` property is added in order to allow use of that subchart, since each subchart injects a value with the same name into the chart values.

The properties `.managementCluster` and `.provider` are added to allow cluster-apps-operator to inject these values.

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.
- [x] Update `/examples` if required.

### CODE REVIEW

**Comment by marians** [comment]
Something is injecting a property `.ssh` on the root level of the values. Where does that happen?

image

**Comment by marians** [comment]
@vxav As you can see, with this PR, the check `JSON schema validation / Verify values.schema.json with schemalint` turns green :)

**Comment by marians** [comment]
Turning this into a draft again, as I found out something new.

File: helm/cluster-cloud-director/values.schema.json [hunk 1]
```diff
@@ -684,8 +719,9 @@
                     "type": "object",
                     "title": "Labels",
                     "description": "These labels are added to the Kubernetes resourses defining this cluster.",
+                    "additionalProperties": false,
                     "patternProperties": {
-                        "^[a-zA-Z0-9\\._-]+$": {
+                        "^[a-zA-Z0-9/\\._-]+$": {
```
Comment (marians) [inline helm/cluster-cloud-director/values.schema.json:724]: Had to add the slash here, which can occur in Kubernetes label keys.

**Review by vxav — APPROVED** [review]
I don't really follow everything in this PR but LGTM if it templates as expected

**Pull request merged**